receiver/pprofreceiver/convert.go

```diff
@@ -0,0 +1,390 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pprofreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/pprofreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/google/pprof/profile"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+	semconv "go.opentelemetry.io/otel/semconv/v1.37.0"
+)
+
+// errInvalPprof is returned for invalid pprof data.
+var errInvalPprof = errors.New("invalid pprof data")
+
+const (
+	// noAttrUnit is a helper to indicate that no
+	// unit is associated to this Attribute.
+	noAttrUnit = int32(-1)
```

Comment:
Can this be a zero? I think with the ["zero index == null" convention](https://github.com/open-telemetry/opentelemetry-proto/pull/711) we've been trying to avoid having to use -1 as a special value.

Reply:
noAttrUnit is just an conversion internal helper and will not be seen in the resulting OTel profiles output. As every positive value can indicate a unit, i needed something to indicate that no unit is set for this attribute.